### DISCUSSION TITLE
Key Fob/ Key Less Sniffing

### DISCUSSION THREAD

**Discussion by Ivanbk** [collin80/SavvyCAN#777] [Q&A]
Are there any examples/documentation on how to sniff car fob codes for reprogramming a key? I would like to emulate the functionality of a commercial key programmer.

**Comment by mitchdetailed** [comment]
reading the debug log of the J2534 system is my recommendation.

**Reply by Ivanbk** [reply]
What is the J2534 system?

**Reply by Ivanbk** [reply]
It seems to be similar to SavvyCAN, logging CAN bus messages

**Reply by collin80** [reply]
J2545 is a standard that is used for automotive programming tools. Essentially it's a standard way to get and receive CAN messages. That way it doesn't matter so much what hardware you actually use. If that hardware has a J2534 driver then it should be able to be supported by a given application for programming the car.
